### PR TITLE
Initial unit-testing implementation

### PR DESCRIPTION
**Pull request by renatgalimov**
I implemented a base for unit-testing. 

Besides that, I updated a few functions that were failing unit tests. 

Unit tests pass on my local emacs 29 and GitHub emacs 27. 

https://github.com/renatgalimov/anki-editor/actions/runs/9023907396

One test fail on emacs 26.

### CODE REVIEW

File: anki-editor.el [hunk 1]
```diff
@@ -790,14 +790,18 @@ and else from variable `anki-editor-prepend-heading'."
...
                                      (car x)
                                      (anki-editor--export-string (cdr x)
                                                                  format)))
                                   fields)))
     (unless deck (user-error "Missing deck"))
     (unless note-type (user-error "Missing note type"))
+
+    ;; Sorting fields not necessary for Anki, but it removes
+    ;; randomness which breaks our tests.
+    (sort exported-fields (lambda (a b) (string< (car a) (car b))))
```
Comment (renatgalimov) [inline anki-editor.el:804]: @slotThe , I know that the speed is important for you. Let me know if you're not ok with this.

Comment (slotThe) [reply anki-editor.el:804]: The `exported-fields` list is tiny—I would be very surprised to have people with more than, say, five fields per card. This seems entirely justified to me if it makes testing easier.

File: test-files/test.org [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+* Cloze with text
+:PROPERTIES:
+:ANKI_NOTE_TYPE: Cloze
+:ANKI_DECK: Tests
+:END:
+** Text
+Lorem {{c1::lorem}}.
+
+* Whatever
+If I remove this - my anki export freezes
```
Comment (slotThe) [inline test-files/test.org:90]: This should hopefully be fixed by https://github.com/anki-editor/anki-editor/commit/c2d66f81382d0c0eb948f67c2cb8152e2b800f59

File: anki-editor.el [hunk 3]
```diff
@@ -1019,15 +1023,15 @@ Return a list of cons of (FIELD-NAME . FIELD-CONTENT)."
                                  "more than two fields missing"))))
       fields)))
 
+
+
```
Comment (renatgalimov) [inline anki-editor.el:1027]:
```suggestion
```

File: anki-editor-tests.el [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+Front content
+
+* Back
+
+Back content
+
+")))
+
+
```
Comment (renatgalimov) [inline anki-editor-tests.el:128]:
```suggestion
```

File: anki-editor-tests.el [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+                                                '(("Front" . "Front content")) 0)))
+    (should (stringp concat-out))
+    (should (equal concat-out
+                   "* Front
+
+Front content
+
+"))))
+
+
```
Comment (renatgalimov) [inline anki-editor-tests.el:141]:
```suggestion
```

File: anki-editor.el [hunk 6]
```diff
@@ -1021,13 +1029,11 @@ Return a list of cons of (FIELD-NAME . FIELD-CONTENT)."
 
 (defun anki-editor--concat-fields (field-names field-alist level)
   "Concat field names and content of fields in list `field-names'."
-  (let ((format (anki-editor-entry-format)))
```
Comment (renatgalimov) [inline anki-editor.el:1024]: It seems that this code has never been used. @slotThe , please let me know if it breaks anything for you.

Comment (slotThe) [reply anki-editor.el:1024]: I've tried this in various ways, and everything seems to work as before :+1:

**Comment by slotThe** [comment]
Thanks!

**Pull request merged**